src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1561,11 +1595,17 @@ static int client_parse_message(
         lease->next_server = message->siaddr;
         lease->address = message->yiaddr;
 
+        if (client->bootp)
+                lease->lifetime = USEC_INFINITY;
```

Comment:
Here's a change suggestion for the missing `server_address` value. I've tested this manually with `bootpd` and it works. Please comment; I'm conflicted over whether `lease->server_address` should be considered at all. In theory, the BOOTP RFC's say "ignore unrecognized options," and if this is not a DHCP request, then we don't know what it means. In practice, if it is present, then we probably do know what it means - and for that matter, if both are present and they differ, we probably know what they both mean. My stance is that it is invalid and should be ignored.

```suggestion
        if (client->bootp) {
                lease->lifetime = USEC_INFINITY;
                lease->server_address = message->siaddr; // || lease->server_address;
                lease->next_server = 0;
        } else
                lease->next_server = message->siaddr;

        lease->address = message->yiaddr;
```

Reply:
I like this, and will apply it to the next version. Thanks!

Reply:
@avramd take a look at 

https://github.com/colin-foster-in-advantage/systemd/commits/add_bootp_staging/

It uses the lease server address if it exists, and falls back to the siaddr.

https://github.com/colin-foster-in-advantage/systemd/commit/efe2577a19184032bf8a336da864936138ae2d93

I also removed the `networkctl renew eth0` issue in that branch. I think this should work for us both... If so, I'll update the docs and re-request a review.